static/js/hpo.js

```diff
@@ -39,95 +39,110 @@ var d3color = d3.scale.linear()
     .domain([-0.2,0,0.2])
     .range(['steelblue','lightgrey','red']);
 
-function transform(input_data, mode, exac_cutoff, cadd_cutoff, exclude_hom, unrelated_flag,hpo_terms,remove_bad_vars){
+function transform(input_data, options){
     // remove_bad_vars determine if dubious variants getting removed.
     // since ppl with the variants marked as dubious have better changes to have the rare variants than average ppl, it's better to remove the patients out of the system. 
     // For dominant cases, if one patient has two qualified variants, but one of them is dubious, when calculating for dominant mode, shall we include this patient?
     // For now, remove the patient completely from the system.
-    // collect qualified patients/vars from HP:0000001
+
+    // what kind of cutoff i'm looking at?
+    var exac = options.mode == 'd' ? 'exac_af' : 'exac_hom';
+
     var qualified_patients = {},
         qualified_vars = {},
+        dubious_vars = {},
         dubious_patients = {};
-    $.each(input_data['HP:0000001'].data, function(k1,v1){
+    //$.each(input_data.data.['HP:0000001'].data, function(k1,v1){
+    // get qualified and dubious vars
+    $.each(input_data.variants, function(k1,v1){
```

Comment:
Ideally where possible try to use the default javascript for loop instead of the `$.each()` - especially when dealing with large inputs - `$.each()` is known to being less efficient...